--- a/src/lib.rs
+++ b/src/lib.rs
@@ -45,7 +45,14 @@
 )]
 #![cfg_attr(
     all(
-        feature = "nightly",
+        any(
+            feature = "nightly",
+            all(
+                target_arch = "wasm32",
+                target_os = "wasi",
+                target_vendor = "wasmer"
+            )
+        ),
         target_family = "wasm",
         target_feature = "atomics"
     ),
--- a/src/thread_parker/mod.rs
+++ b/src/thread_parker/mod.rs
@@ -67,7 +67,14 @@
         #[path = "sgx.rs"]
         mod imp;
     } else if #[cfg(all(
-        feature = "nightly",
+        any(
+            feature = "nightly",
+            all(
+                target_arch = "wasm32",
+                target_os = "wasi",
+                target_vendor = "wasmer"
+            )
+        ),
         target_family = "wasm",
         target_feature = "atomics"
     ))] {
